src/sentry/tasks/integrations/github/open_pr_comment.py

```diff
@@ -193,13 +194,21 @@ def safe_for_comment(
     return safe_to_comment, filtered_pr_files
 
 
-def get_pr_filenames(pr_files: JSONData) -> List[str]:
+def get_pr_filenames_and_patches(pr_files: JSONData) -> Tuple[List[str], List[str]]:
     # new files will not have sentry issues associated with them
     # only fetch Python files
     pr_filenames: List[str] = [file["filename"] for file in pr_files]
+    patches: List[str] = [file["patch"] for file in pr_files]
 
     logger.info("github.open_pr_comment.pr_filenames", extra={"count": len(pr_filenames)})
-    return pr_filenames
+
+    return pr_filenames, patches
+
+
+# currently Python only
+def get_file_functions(patch: str) -> Set[str]:
+    python_function_regex = r"^@@.*@@\s+def\s+(?P<fnc>.*)\(.*$"
```

Comment:
Can you add some description of what this regex does, similar to:
https://github.com/getsentry/sentry/blob/e7e6d3909f1074520c8a4f57f2d8463d98ba0794/src/sentry/api/fields/sentry_slug.py#L20-L27

Reply:
😵‍💫 will an example suffice

Reply:
chatgpt spits out a nice explanation, that's how I got the one above. It's not too bad

Reply:
It's just nice to have cuz otherwise you come back to it and have no clue what it does